src/mlpack/methods/random_forest/random_forest_main.cpp

```diff
@@ -201,10 +211,17 @@ static void mlpackMain()
   ReportIgnoredParam({{ "training", false }}, "minimum_leaf_size");
 
   RandomForestModel* rfModel;
+  // Handles the case when we are either training on top of existing forest
+  // else we are making predictions only.
+  if (IO::HasParam("warm_start") or IO::HasParam("input_model"))
```

Comment:
I think we are checking earlier that if `warmStart` is passed then `input_model` has to be passed. So checking on `input_model` is sufficient here.

Reply:
Yes you are correct. Checking for `warm_start` is redundant here. But the reason I put it here is for the documentation for future developers. Let me explain you with in the following manner.

There are following possible cases how a person will use this binding :-
```
Purpose                             training    input_model     warm_start
Train from scratch                    yes            no              no
Train on top of existing              yes            yes            yes
Make prediction on pre-trained         no            yes             no
```

Thus to handle the bottom two cases, I have added this if statement. 
Now, the reason for this condition might not be very straightforward for someone to understand if someone doesn't think in the above explained manner. Thus to make it explicitly clear, I have put both these conditions.

Hope my thought process for making this choice make sense. Let me know what you think about it :-)

Reply:
Personally my take is that the old approach was fine.  Maybe we can stick with that and just add a comment to the line where we load the `input_model` that we could be loading it for either warm-started training or prediction (or both)?